client/queue.go

```diff
@@ -139,8 +144,10 @@ func newInstanceQueue(instance placement.Instance, opts Options) instanceQueue {
 	}
 	q.writeFn = q.conn.Write
 
+	q.wg.Add(2)
```

Comment:
What do you think of using an anonymous function to make it clear when `wg.Done` is closed? For example:

```
go func() {
  q.drain()
  q.wg.Done()
}
```

Reply:
Talked offline, will keep it as is to be consistent with other parts of the code base.